infra/lambda/datasets-endpoint/test_function.py

```diff
@@ -0,0 +1,307 @@
+"""
+Basic Lambda function tests. Working Data Lake AWS environment is required
+(run '$ cdk deploy' before running tests).
+"""
+
+import logging
+import re
+import uuid
+
+import function  # pylint:disable=import-error
+import pytest
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def test_request_no_method():
+    """Try request not containing method."""
+
+    body = {}
+
+    resp = function.lambda_handler({"body": body}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'httpMethod' is a required property."
+
+
+def test_request_no_body():
+    """Try request not containing body."""
+
+    method = "POST"
+
+    resp = function.lambda_handler({"httpMethod": method}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'body' is a required property."
+
+
+def test_post_method(db_truncate):  # pylint:disable=unused-argument
+    """Test Dataset creation using POST method."""
+
+    method = "POST"
+    body = {}
+    body["type"] = "RASTER"
+    body["title"] = "Dataset {}".format(uuid.uuid4().hex[:8])
+    body["owning_group"] = "A_XYZ_XYZ"
+
+    resp = function.lambda_handler({"httpMethod": method, "body": body}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 201
+    assert resp["body"]["type"] == body["type"]
+    assert resp["body"]["title"] == body["title"]
+    assert resp["body"]["owning_group"] == body["owning_group"]
+
+    pytest.dataset_id = resp["body"]["id"]
```

Comment:
This is not a good idea:

1. Tests will fail if they are run in a different sequence, which we probably want to introduce ASAP to avoid test dependencies.
1. It adds properties to a third party module. The properties don't belong to that module, but even worse they may actually override real properties on that module, which would break its functionality in a way which would probably be hard to debug.
1. Tests can no longer run individually. Running a single test after a change can drastically speed up development, especially when running expensive tests like these ones.

The standard pattern for expensive setup is the standard Python `unittest` pattern of a `setUpClass` method which creates everything needed by every test, assigning the relevant values to test class properties.

Reply:
I agree, I was already thinking about rewriting tests to be independent using setup and teardown methods. It is faster to run one test, but significantly slower to run all tests. This change is already pushed in 3f1abbaa1270dddaf8aa9620eb1ac0d5e4371599

Reply:
I'm not sure what you mean. `setUpClass` runs once per test class, so it should be much faster than doing setup and teardown for each test function.

Reply:
For now "DB Setup" and "Teardown" are running for each test to make them independent. This is of course slower in case of running all tests.

Reply:
Ok, let's leave it for now. The test suite will get much slower over time, but it should be easy to fix.